### PR TITLE
app-emulation/libvirt-glib: Version updated to 4.0.0

### PR DESCRIPTION
**Pull request by jpds**
Tested on `amd64`

### CODE REVIEW

File: app-emulation/libvirt-glib/libvirt-glib-4.0.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+SRC_URI="https://libvirt.org/sources/glib/${P}.tar.xz"
+
+LICENSE="LGPL-2.1"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~ppc64 ~x86"
+IUSE="+introspection +vala"
+REQUIRED_USE="vala? ( introspection )"
+
+BDEPEND="
+        dev-util/glib-utils
```
Comment (mattst88) [inline app-emulation/libvirt-glib/libvirt-glib-4.0.0.ebuild:19]: Use tabs for indentation, and only one per level.

File: app-emulation/libvirt-glib/libvirt-glib-4.0.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        dev-util/gtk-doc-am
+        >=dev-util/intltool-0.35.0
+        virtual/pkgconfig
+        vala? ( $(vala_depend) )"
```
Comment (mattst88) [inline app-emulation/libvirt-glib/libvirt-glib-4.0.0.ebuild:23]: I like to put the trailing `"` on its own line. It helps minimize the diff if a dependency is added in the future.

File: app-emulation/libvirt-glib/libvirt-glib-4.0.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+RDEPEND="
+        dev-libs/libxml2:2
+        >=app-emulation/libvirt-1.2.6:=
+        >=dev-libs/glib-2.38.0:2
```
Comment (mattst88) [inline app-emulation/libvirt-glib/libvirt-glib-4.0.0.ebuild:27]: `meson.build` says the minimum required is 2.48.

File: app-emulation/libvirt-glib/libvirt-glib-4.0.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+BDEPEND="
+        dev-util/glib-utils
+        dev-util/gtk-doc-am
+        >=dev-util/intltool-0.35.0
+        virtual/pkgconfig
+        vala? ( $(vala_depend) )"
+RDEPEND="
+        dev-libs/libxml2:2
+        >=app-emulation/libvirt-1.2.6:=
```
Comment (mattst88) [inline app-emulation/libvirt-glib/libvirt-glib-4.0.0.ebuild:26]: `meson.build` says the minimum required is 1.2.8.

File: app-emulation/libvirt-glib/libvirt-glib-4.0.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        >=dev-libs/glib-2.38.0:2
+        introspection? ( >=dev-libs/gobject-introspection-1.36.0:= )"
```
Comment (mattst88) [inline app-emulation/libvirt-glib/libvirt-glib-4.0.0.ebuild:28]: `meson.build` says the gobject-intospection minimum required is the same as the glib version (2.48).

File: app-emulation/libvirt-glib/libvirt-glib-4.0.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+DEPEND="${BDEPEND}
+        ${RDEPEND}"
+
+src_prepare() {
+        default
```
Comment (mattst88) [inline app-emulation/libvirt-glib/libvirt-glib-4.0.0.ebuild:34]: Single tab for indentation.

File: app-emulation/libvirt-glib/libvirt-glib-4.0.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+REQUIRED_USE="vala? ( introspection )"
+
+BDEPEND="
+        dev-util/glib-utils
+        dev-util/gtk-doc-am
+        >=dev-util/intltool-0.35.0
+        virtual/pkgconfig
+        vala? ( $(vala_depend) )"
+RDEPEND="
+        dev-libs/libxml2:2
```
Comment (mattst88) [inline app-emulation/libvirt-glib/libvirt-glib-4.0.0.ebuild:25]: `meson.build` says the minimum required is 2.9.1.

File: app-emulation/libvirt-glib/libvirt-glib-4.0.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+src_prepare() {
+        default
+        use vala && vala_src_prepare
+}
+
+src_configure() {
+        local emesonargs=(
+                $(meson_feature introspection)
+                $(meson_feature vala vapi)
```
Comment (mattst88) [inline app-emulation/libvirt-glib/libvirt-glib-4.0.0.ebuild:41]: `meson_options.txt` has two other configuration options that I think we should have:
````
option('docs', type: 'feature', value: 'auto', description: 'whether to generate documentation')
option('tests', type: 'feature', value: 'auto', description: 'whether to build tests')
````

`docs` should be controlled by an `IUSE=gtk_doc` flag and should depend on `dev-util/gtk-doc`.

`tests` should be controlled by an `IUSE=test` flag that is implicitly enabled when building with `FEATURES=test`.

**Review by mattst88 — CHANGES_REQUESTED** [review]

**Comment by jpds** [comment]
@mattst88 Thanks for the review - I've fixed everything (my editor oddly added a space to my tabs), but the gobject-intospection dependency as 2.48 for it doesn't exist.

**Comment by mattst88** [comment]
> @mattst88 Thanks for the review - I've fixed everything (my editor oddly added a space to my tabs), but the gobject-intospection dependency as 2.48 for it doesn't exist.

Thanks! I meant *1*.48. Sorry about the confusion. Looks like you figured out my mistake :)

**Comment by mattst88** [comment]
Thanks!

There are tons of needed GNOME version bumps (https://packages.gentoo.org/maintainer/gnome@gentoo.org/outdated). If you're interested in handling any others, please keep them coming :)

**Pull request closed without merging**